api/v1alpha2/conversion_test.go

```diff
@@ -24,6 +24,21 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/utils/pointer"
 	"sigs.k8s.io/cluster-api/api/v1alpha3"
+	bootstrapv1a2 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha2"
+	bootstrapv1a3 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha3"
+)
+
+var (
+	kubeAdmObj = &corev1.ObjectReference{
+		Kind:       kubeadmConfigKind,
+		APIVersion: bootstrapv1a2.GroupVersion.String(),
+	}
+	kubeAdmConfigTemplateObj = &corev1.ObjectReference{
+		Kind:       kubeadmConfigTemplateKind,
+		APIVersion: bootstrapv1a2.GroupVersion.String(),
+	}
+	verV1alpha2 = bootstrapv1a2.GroupVersion.String()
+	verV1alpha3 = bootstrapv1a3.GroupVersion.String()
 )
```

Comment:
We can't have the object references as package level variables. If you use them with the objects being tested, they get modified by the conversion code, and the tests "pass" but they're not testing the correct values.